src/cargo/core/summary.rs

```diff
@@ -403,25 +356,35 @@ impl FeatureValue {
         }
     }
 
-    /// Returns `true` if this feature explicitly used `dep:` syntax.
-    pub fn has_dep_prefix(&self) -> bool {
-        matches!(self, FeatureValue::Dep { .. })
+    fn explicitly_name(&self) -> Option<InternedString> {
+        match self {
+            FeatureValue::Dep { dep_name, .. } => Some(*dep_name),
+            _ => None,
+        }
+    }
+
+    fn dep_name(&self) -> InternedString {
```

Comment:
I think this might be a confusing name, since `FeatureValue::Feature` is not necessarily a dependency name. For example:

```toml
[features]
foo = ["bar"]
bar = []
```

Here the `FeatureValue::Feature("bar")` is not a dependency name.

I realize this is refactoring code that was already using `dep_name`, but the original was somewhat abusing the nomenclature.

I would go with `feature_or_dep_name` to make it clearer it can be either. What do you think?